### ISSUE TITLE
misbehaviour of helm upgrade --install when adding --dry-run

### ISSUE THREAD

**Issue by mvswahn** [helm/helm#4437]
When installing a new service via helm with the following command:

```
helm upgrade --install --reset-values --recreate-pods dev-message-broker stable/rabbitmq
```

The release is created and everything works fine.
```
Release "dev-message-broker" does not exist. Installing it now.
```

Although, when I do the same, just with `--dry-run` added, this happens:

```
helm upgrade --install --reset-values --recreate-pods dev-message-broker stable/rabbitmq --dry-run
```

Output:

```
Error: UPGRADE FAILED: "dev-message-broker" has no deployed releases
```

I'd expect that the --dry-run does not fail, since --install will clearly install the service.

Output of `helm version`:

Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:13:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:13:31Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
AWS EC2 / Amazon Linux AMI

**Comment by bacongobbler** [comment]
This was fixed in master and will be available in the 2.10 release

**Comment by bacongobbler** [comment]
#4402 for reference
